refactor(transformer/styled-components): simplify concating strs

Dunqing opened the pull request:
`atom_from_strs_array` returns an `Atom`, but we only need a `str` here, so using `alloc_concat_strs_array` is better.

overlookmotel commented:
### Merge activity

* **Jul 10, 8:49 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Dunqing committed:
refactor(transformer/styled-components): simplify concating strs (#12183)

`atom_from_strs_array` returns an `Atom`, but we only need a `str` here, so using `alloc_concat_strs_array` is better.
